File: upl-purchase-sdk-java/src/main/java/com/unzer/paylater/model/Person.java
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */
package com.unzer.paylater.model;

import java.time.LocalDate;
import java.util.Objects;

/**
 * Properties of a natural person.
 */
public class Person {

    private String salutation;
    private String firstName;
    private String lastName;
    private LocalDate birthdate;
    private String socialId;
    private Occupation occupation;

    /**
     * @return salutation
     */
    public String getSalutation() {
        return salutation;
    }
    public void setSalutation(String salutation) {
        this.salutation = salutation;
    }
    public Person withSalutation(String salutation) {
        this.salutation = salutation;
        return this;
    }

    /**
     * @return firstName
     */
    public String getFirstName() {
        return firstName;
    }
    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }
    public Person withFirstName(String firstName) {
        this.firstName = firstName;
        return this;
    }

    /**
     * @return lastName
     */
    public String getLastName() {
        return lastName;
    }
    public void setLastName(String lastName) {
        this.lastName = lastName;
    }
    public Person withLastName(String lastName) {
        this.lastName = lastName;
        return this;
    }

    /**
     * @return birthdate
     */
    public LocalDate getBirthdate() {
        return birthdate;
    }
    public void setBirthdate(LocalDate birthdate) {
        this.birthdate = birthdate;
    }
    public Person withBirthdate(LocalDate birthdate) {
        this.birthdate = birthdate;
        return this;
    }

    /**
     * @return socialId
     */
    public String getSocialId() {
        return socialId;
    }
    public void setSocialId(String socialId) {
        this.socialId = socialId;
    }
    public Person withSocialId(String socialId) {
        this.socialId = socialId;
        return this;
    }

    /**
     * @return occupation
     */
    public Occupation getOccupation() {
        return occupation;
    }
    public void setOccupation(Occupation occupation) {
        this.occupation = occupation;
    }
    public Person withOccupation(Occupation occupation) {
        this.occupation = occupation;
        return this;
    }

    @Override
    public boolean equals(java.lang.Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        Person person = (Person) o;
        return Objects.equals(this.salutation, person.salutation)
                && Objects.equals(this.firstName, person.firstName)
                && Objects.equals(this.lastName, person.lastName)
                && Objects.equals(this.birthdate, person.birthdate)
                && Objects.equals(this.socialId, person.socialId)
                && Objects.equals(this.occupation, person.occupation);
    }

    @Override
    public int hashCode() {
        return Objects.hash(salutation, firstName, lastName, birthdate, socialId, occupation);
    }

    @Override
    public String toString() {
        return "class Person {\n"
                + "        salutation: " + toIndentedString(salutation) + "\n"
                + "        firstName: " + toIndentedString(firstName) + "\n"
                + "        lastName: " + toIndentedString(lastName) + "\n"
                + "        birthdate: " + toIndentedString(birthdate) + "\n"
                + "        socialId: " + toIndentedString(socialId) + "\n"
                + "        occupation: " + toIndentedString(occupation) + "\n"
                + "}";
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces
     * (except the first line).
     */
    private String toIndentedString(java.lang.Object o) {
        return o == null
                ? "null"
                : o.toString().replace("\n", "\n        ");
    }
}

File: upl-purchase-sdk-js/src/unzer-paylater/model/Person.js
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */

const ModelHelper = require("../util/ModelHelper");
const Occupation = require("./Occupation");

/**
 * Properties of a natural person.
 */
class Person {
    /**
     * @returns { String }
     */
    getSalutation() {
        return this.salutation;
    }
    /**
     * @param { String } salutation
     */
    setSalutation(salutation) {
        this.salutation = ModelHelper.validatePrimitive(salutation, "string");
    }
    /**
     * @param { String } val
     */
    withSalutation(val) {
        this.setSalutation(val);
        return this;
    }

    /**
     * @returns { String }
     */
    getFirstName() {
        return this.firstName;
    }
    /**
     * @param { String } firstName
     */
    setFirstName(firstName) {
        this.firstName = ModelHelper.validatePrimitive(firstName, "string");
    }
    /**
     * @param { String } val
     */
    withFirstName(val) {
        this.setFirstName(val);
        return this;
    }

    /**
     * @returns { String }
     */
    getLastName() {
        return this.lastName;
    }
    /**
     * @param { String } lastName
     */
    setLastName(lastName) {
        this.lastName = ModelHelper.validatePrimitive(lastName, "string");
    }
    /**
     * @param { String } val
     */
    withLastName(val) {
        this.setLastName(val);
        return this;
    }

    /**
     * @returns { Date }
     */
    getBirthdate() {
        return new Date(this.birthdate);
    }
    /**
     * @param { Date } birthdate
     */
    setBirthdate(birthdate) {
        this.birthdate = ModelHelper.validateDate(birthdate);
    }
    /**
     * @param { Date } val
     */
    withBirthdate(val) {
        this.setBirthdate(val);
        return this;
    }

    /**
     * @returns { String }
     */
    getSocialId() {
        return this.socialId;
    }
    /**
     * @param { String } socialId
     */
    setSocialId(socialId) {
        this.socialId = ModelHelper.validatePrimitive(socialId, "string");
    }
    /**
     * @param { String } val
     */
    withSocialId(val) {
        this.setSocialId(val);
        return this;
    }

    /**
     * @returns { Occupation }
     */
    getOccupation() {
        return this.occupation;
    }
    /**
     * @param { Occupation } occupation
     */
    setOccupation(occupation) {
        this.occupation = ModelHelper.validateObject(occupation, Occupation);
    }
    /**
     * @param { Occupation } val
     */
    withOccupation(val) {
        this.setOccupation(val);
        return this;
    }

    /**
     * @returns { Person }
     */
    static constructFromObject(data) {
        if (!data) return undefined;
        return new Person()
            .withSalutation(ModelHelper.convertToType(data["salutation"], String))
            .withFirstName(ModelHelper.convertToType(data["firstName"], String))
            .withLastName(ModelHelper.convertToType(data["lastName"], String))
            .withBirthdate(ModelHelper.convertToType(data["birthdate"], Date))
            .withSocialId(ModelHelper.convertToType(data["socialId"], String))
            .withOccupation(ModelHelper.convertToType(data["occupation"], Occupation));
    }
}

module.exports = Person;
